wal/wal.proto

```diff
@@ -0,0 +1,19 @@
+syntax = "proto3";
+
+package wal;
+option go_package = "internal/wal";
+
+enum EntryType {
```

Comment:
Let's call this `WALEntryType`, prefixing it with `WAL` makes the purpose clearer.